docs(refManualReset): add deep reactivity usage note

xiaoland opened the pull request:
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.

### Description

Fixes #5200

xiaoland committed:
docs: refManualReset deep reactivity usage note

9romise commented on packages/shared/refManualReset/index.md:
```diff
@@ -19,3 +19,32 @@ message.reset()
+    <button @click="increaseAge">Add age</button>
+  </div>
+</template>
+
```
There's a missing ``` here.

xiaoland replied:
Addressed in e18f9dd46f5bfb26157fdc6d5d999b9de70ba932

xiaoland committed:
docs(refManualReset): add missing end tag of literal block

OrbisK commented:
I am not sure if this is what we want for our ref composables. Passing a ref/reactive to `ref()` does normally return the ref/reactive itself directly. Perhaps we should consider a option to make them deep. 🤔

OrbisK commented:
I see. However, I think we should only document that this function is shallow, rather than the reactive workaround.

xiaoland committed:
docs(refManualReset): remove reactive workaround

Removed example code for using reactive with refManualReset and clarified its shallow reactivity.

xiaoland commented:
> I see. However, I think we should only document that this function is shallow, rather than the reactive workaround.

I totally agree, add a workaround in official document might confuse or mislead users since this approach might not be the best choice for user. 
Therefore I have made changes at commit 9c0da39cdfed237ad7a2a98a10f1231eb2e225f1 , please review it, thank you.

OrbisK committed:
Merge branch 'main' into doc/ref-manual-reset-deep-reactivity

OrbisK committed:
docs: write note as callout

OrbisK committed:
Merge branch 'main' into doc/ref-manual-reset-deep-reactivity
